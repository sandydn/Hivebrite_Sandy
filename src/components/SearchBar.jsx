import React from 'react';
import styled from 'styled-components';
import logo from '../assets/SearchIcon.svg';

const InputWrapper = styled.div`
  width: 300px;
  display: flex;
  justify-content: center;
`;

const Input = styled.input`
  line-height: 4.7rem;
  font-size: 1.3rem;
  background: none;
  padding: 5px 0px 0px 5px;
  border-style: none;
  outline: none;
  text-rendering: auto;
  caret-color: white;
  ::placeholder {
    color: white;
  },
  field__prefix_active {
    color: white;
    font-size: 1.3rem;
  } 
`;

const SearchBar = props => { 
  const {handleSearch} = props;
  return (
      <InputWrapper>
        <img src={logo} alt='search-icon'/>
        <Input type="text" placeholder='Enter a city name' onChange={handleSearch} />
      </InputWrapper>
  )
}

export default SearchBar;