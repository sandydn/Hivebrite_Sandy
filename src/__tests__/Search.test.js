import * as React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import SearchBar from '../components/SearchBar.jsx';

describe('SearchBar', () => {
    const onHandleClickMock = jest.fn();
    const props = { handleSearch: () => onHandleClickMock('Arg')};
    const wrapper = shallow(<SearchBar {...props} />)

    describe('Should pass on start.', () => {
        it('Should render correctly.', () => {
            const tree = renderer
                .create(<SearchBar {...props} />)
                .toJSON();
        expect(tree).toMatchSnapshot();
        });

        it('Should call "handleSearch" when write a city name.', () => {
            wrapper.find("[type='text']").simulate('change');
        expect(onHandleClickMock).toHaveBeenCalled();
        });
    });
});