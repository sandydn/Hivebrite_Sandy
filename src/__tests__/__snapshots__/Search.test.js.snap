// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Should pass on start. Should render correctly. 1`] = `
<div
  className="sc-AxjAm bnZBsp"
>
  <img
    alt="search-icon"
    src="SearchIcon.svg"
  />
  <input
    className="sc-AxirZ hMBrHq"
    onChange={[Function]}
    placeholder="Enter a city name"
    type="text"
  />
</div>
`;
